### ISSUE TITLE
Request for paper evaluation pipeline: URaG-3B DUDE reproduction (51.45 vs. 54.1 ANLS), prompts, post-processing, and submission format

### ISSUE THREAD

**Issue by FFFFFFUN** [shi-yx/URaG#5]
<html><head></head><body><p><span>Title: Request for paper evaluation pipeline: URaG-3B DUDE reproduction (51.45 vs. 54.1 ANLS), prompts, post-processing, and submission format</span></p><p><span>Hi URaG authors,</span></p><p><span>Thank you for releasing URaG and the model checkpoints. We are evaluating URaG-3B as a baseline for further research and would like to reproduce the paper results before building on it.</span></p><p><span>Our main unresolved discrepancy is on </span><strong><span>DUDE</span></strong><span>. Using the released checkpoint and an inference runner based on the public demo, our latest official test result is </span><strong><span>51.45 ANLS</span></strong><span>, compared with </span><strong><span>54.1</span></strong><span> reported in the paper.</span></p><p><span>Could you release the </span><strong><span>actual end-to-end evaluation scripts and configurations used for the paper</span></strong><span>, particularly for DUDE? We need to establish whether the remaining difference comes from our implementation, an undocumented evaluation setting, or another source.</span></p><h2><span>1. Results obtained so far</span></h2><p><span>All values below are on a 0–100 scale.</span></p>
Benchmark | Metric | Paper URaG-3B | Our result | Evaluation route
-- | -- | -- | -- | --
SlideVQA test | EM | 63.8 | 63.70 | Local evaluation
MP-DocVQA test | ANLS | 86.0 | 85.06 | Official RRC submission
DUDE test | ANLS | 54.1 | 51.45 | Official RRC submission

<p><span>The fixed parser maps exact </span><code><span>none</span></code><span> to an empty answer and otherwise splits only on </span><code><span>|</span></code><span>.</span></p><p><span>The separate validation experiment explicitly requested pipe-separated list items and </span><code><span>none</span></code><span> for unanswerable questions. It improved list and unanswerable scores but reduced extractive/abstractive performance, with no overall improvement. We therefore did </span><strong><span>not</span></strong><span> adopt this prompt for the test evaluation.</span></p><p><span>This was a validation diagnostic, not your claimed original evaluation protocol. It also used three workers rather than four, so we do not claim strict numerical determinism across runs.</span></p><p><span>A relevant observation: among the </span><strong><span>467 validation list questions</span></strong><span>, </span><strong><span>325 original responses contain comma-space</span></strong><span>, and none contains </span><code><span>|</span></code><span>. Some responses contain the correct answer items in a single string, whereas others genuinely omit items or give incorrect answers. Therefore, we cannot attribute all list errors to serialization, nor safely apply unconditional comma splitting.</span></p><p><strong><span>What output convention did you use during DUDE training and evaluation, and how were list boundaries recovered without access to test answer types?</span></strong></p><h2><span>5. Data and completeness checks</span></h2><p><span>We checked the following:</span></p><ul><li><p><span>Test: 11,402 predictions, all successful, with unique IDs exactly matching the official submission template.</span></p></li><li><p><span>Validation: 6,318 rows; one duplicate question ID also exists in the source annotations and is retained in our row-wise audit.</span></p></li><li><p><span>Question IDs, questions, and document mappings match the prepared manifests.</span></p></li><li><p><span>No missing image paths or page-number ordering gaps were found in those manifests.</span></p></li><li><p><span>Recorded retrieval-score lengths match the number of input pages.</span></p></li><li><p><span>There is no broad indication that the 256-token generation limit truncated the outputs.</span></p></li></ul><p><span>We used DUDE annotations reporting version </span><strong><span>1.0.8</span></strong><span> and the distributed page-image archive.</span></p><p><span>We also found two documents whose distributed image sets contain only 50 pages although their PDFs are longer: one validation document and one test document. Archive inspection confirmed that these image counts originate in the downloaded package, rather than local page truncation. The test case affects seven questions, so it cannot explain most of the score gap.</span></p><p><strong><span>Which DUDE annotation/image version did you use? Did you use the distributed images or render the complete PDFs, and were any page limits, filtering rules, or excluded question IDs applied?</span></strong></p><h2><span>6. Requested release</span></h2><p><span>The most useful artifact would be the </span><strong><span>paper experiment pipeline</span></strong><span>, rather than another single-document demo or a reference to the metric formula alone:</span></p><ul><li><p><span>Dataset preparation and exact split/version selection.</span></p></li><li><p><span>Benchmark-specific prompts and image preprocessing.</span></p></li><li><p><span>Batch inference scripts and complete generation configurations.</span></p></li><li><p><span>List/no-answer extraction and post-processing.</span></p></li><li><p><span>Official submission export scripts.</span></p></li><li><p><span>Evaluation commands and expected outputs.</span></p></li><li><p><span>Checkpoint revision corresponding to the reported results.</span></p></li><li><p><span>Ideally, the DUDE test prediction/submission file underlying the reported </span><strong><span>54.1 ANLS</span></strong><span>, if shareable.</span></p></li></ul><p><span>DUDE is our immediate priority. Equivalent scripts for MP-DocVQA, SlideVQA, and MMLongBench-Doc would also help establish a consistent reproduction baseline. For retrieval results, please specify the evaluated split/subset and the exact Top-1/Top-5 definition for questions with multiple evidence pages.</span></p><p><span>We have retained raw predictions, configurations, hashes, submission files, and per-category results, and can provide targeted examples or logs.</span></p><p><span>Our goal is to reproduce the published evaluation faithfully—not to tune submission formatting against hidden test scores. If some experimental settings differ from the public demo or checkpoint defaults, documenting those differences would be especially helpful.</span></p><p><span>Thank you!</span></p></body></html>Title: Request for paper evaluation pipeline: URaG-3B DUDE reproduction (51.45 vs. 54.1 ANLS), prompts, post-processing, and submission format

Hi URaG authors,

Thank you for releasing URaG and the model checkpoints. We are evaluating URaG-3B as a baseline for further research and would like to reproduce the paper results before building on it.

Our main unresolved discrepancy is on **DUDE**. Using the released checkpoint and an inference runner based on the public demo, our latest official test result is **51.45 ANLS**, compared with **54.1** reported in the paper.

Could you release the **actual end-to-end evaluation scripts and configurations used for the paper**, particularly for DUDE? We need to establish whether the remaining difference comes from our implementation, an undocumented evaluation setting, or another source.

## 1. Results obtained so far

All values below are on a 0–100 scale.

| Benchmark | Metric | Paper URaG-3B | Our result | Evaluation route |
|---|---|---:|---:|---|
| SlideVQA test | EM | 63.8 | 63.70 | Local evaluation |
| MP-DocVQA test | ANLS | 86.0 | 85.06 | Official RRC submission |
| DUDE test | ANLS | 54.1 | 51.45 | Official RRC submission |

The SlideVQA result is close to the paper, but we do not regard numerical agreement alone as proof that every evaluation detail matches.

### DUDE official test breakdown

| Category | Original submission | Current submission |
|---|---:|---:|
| Overall ANLS | 47.09 | 51.45 |
| Extractive | 65.03 | 65.03 |
| Abstractive | 61.12 | 60.54 |
| List of answers | 9.92 | 9.92 |
| Unanswerable | 0.00 | 24.85 |

These two submissions use **exactly the same model predictions**. The current submission converts exact case-insensitive `None` responses into empty strings; all other answer strings remain unchanged.

There are **11,402 test predictions**, including **721 `None` responses**. This conversion explains part of the original gap, but the remaining gap is **2.65 ANLS points**, and list-answer performance remains low.

We recognize that our original no-answer handling was an evaluation issue on our side. We are not treating the original 47.09 result as evidence against the released model.

## 2. Checkpoint and inference settings

- Model: [`[shiyx1/URaG-3B](https://huggingface.co/shiyx1/URaG-3B)`](https://huggingface.co/shiyx1/URaG-3B)
- Official source commit used for comparison: `44d980369751cd02d2411edbe322414261d21d1a`
- Python 3.10
- PyTorch `2.5.1+cu124`
- Transformers `4.49.0`
- FlashAttention `2.7.3`
- BF16 and `flash_attention_2`
- Four RTX 4090 workers, one model replica per GPU, processing disjoint question shards

Our runner follows the public demo for:

- Page images before the question in the user message.
- The processor’s chat template and generation prompt.
- `min_pixels=100352`, `max_pixels=802816`.
- `remain_pages=5`.
- `layer_for_retrieval=6`.
- `max_new_tokens=256`.
- The question-position offset derived from the suffix token length.
- Removing input tokens before decoding the generated answer.

The exact suffix is:

```text
\nPlease try to answer the question with short words or phrases if possible.
```

All available page images in the manifest are passed to the model. Ground-truth answers, answer types, and evidence pages are **not** passed to inference.

Our model-code changes for the original run were diagnostic additions to record retrieval scores and page indices; static comparison did not show replacement of the original retrieval or generation path.

### Generation configuration

The loaded checkpoint configuration is:

```json
{
  "do_sample": true,
  "temperature": 0.1,
  "top_k": 1,
  "top_p": 0.001,
  "repetition_penalty": 1.05
}
```

The paper describes `temperature=1`, `top_p=1`, and seed 42. Our original runner did not explicitly override the seed.

**Which effective generation configuration produced the reported results?** In particular, were `do_sample`, `top_k`, temperature, and top-p overridden during benchmark evaluation?

## 3. DUDE submission interface ambiguity

The current RRC submission instructions and official sample use:

```json
{
  "questionId": "...",
  "answer": "model answer",
  "answer_confidence": 1.0,
  "answer_abstain": false
}
```

However, the standalone [`[Jordy-VL/DUDEeval](https://github.com/Jordy-VL/DUDEeval)`](https://github.com/Jordy-VL/DUDEeval) scorer expects an `answers` array.

We confirmed that the current online endpoint rejects files without `answer_abstain`, and also rejects files containing only plural `answers` instead of singular `answer`. The official sample contains placeholder strings, without examples of list or unanswerable outputs.

Could you clarify:

1. Did you obtain the paper’s DUDE test score through this RRC endpoint, or through another evaluation package/version?
2. How did you serialize **list answers** into the online submission?
3. How did you represent **unanswerable answers**?
4. Was any answer extraction, normalization, or benchmark-specific post-processing applied before submission?
5. Could you provide an actual submission example covering single-answer, list-answer, and unanswerable questions?

Our confidence and abstention fields currently retain constant placeholders. We understand that OOD abstention is distinct from a question being unanswerable from its document.

## 4. Validation-set diagnostics

We audited the existing **6,318 validation predictions** with the official DUDEeval scoring functions, pinned to commit:

```text
8bd6deac0f9747e344338e3850f02bf76744d888
```

We corrected an earlier local aggregation issue that excluded questions with empty ground-truth answers. All questions are now retained in the denominator. Ground-truth answer types select the scoring formula, but are not supplied to prediction parsing.

The validation results are:

| Setting | Full-val ANLS |
|---|---:|
| Original predictions, each response treated as one answer string | 48.712 |
| Original predictions with fixed `pipe_none_v1` post-processing | 52.095 |
| Separate list/no-answer instruction experiment, same fixed parser | 51.847 |

The fixed parser maps exact `none` to an empty answer and otherwise splits only on `|`.

The separate validation experiment explicitly requested pipe-separated list items and `none` for unanswerable questions. It improved list and unanswerable scores but reduced extractive/abstractive performance, with no overall improvement. We therefore did **not** adopt this prompt for the test evaluation.

This was a validation diagnostic, not your claimed original evaluation protocol. It also used three workers rather than four, so we do not claim strict numerical determinism across runs.

A relevant observation: among the **467 validation list questions**, **325 original responses contain comma-space**, and none contains `|`. Some responses contain the correct answer items in a single string, whereas others genuinely omit items or give incorrect answers. Therefore, we cannot attribute all list errors to serialization, nor safely apply unconditional comma splitting.

**What output convention did you use during DUDE training and evaluation, and how were list boundaries recovered without access to test answer types?**

## 5. Data and completeness checks

We checked the following:

- Test: 11,402 predictions, all successful, with unique IDs exactly matching the official submission template.
- Validation: 6,318 rows; one duplicate question ID also exists in the source annotations and is retained in our row-wise audit.
- Question IDs, questions, and document mappings match the prepared manifests.
- No missing image paths or page-number ordering gaps were found in those manifests.
- Recorded retrieval-score lengths match the number of input pages.
- There is no broad indication that the 256-token generation limit truncated the outputs.

We used DUDE annotations reporting version **1.0.8** and the distributed page-image archive.

We also found two documents whose distributed image sets contain only 50 pages although their PDFs are longer: one validation document and one test document. Archive inspection confirmed that these image counts originate in the downloaded package, rather than local page truncation. The test case affects seven questions, so it cannot explain most of the score gap.

**Which DUDE annotation/image version did you use? Did you use the distributed images or render the complete PDFs, and were any page limits, filtering rules, or excluded question IDs applied?**

## 6. Requested release

The most useful artifact would be the **paper experiment pipeline**, rather than another single-document demo or a reference to the metric formula alone:

- Dataset preparation and exact split/version selection.
- Benchmark-specific prompts and image preprocessing.
- Batch inference scripts and complete generation configurations.
- List/no-answer extraction and post-processing.
- Official submission export scripts.
- Evaluation commands and expected outputs.
- Checkpoint revision corresponding to the reported results.
- Ideally, the DUDE test prediction/submission file underlying the reported **54.1 ANLS**, if shareable.

DUDE is our immediate priority. Equivalent scripts for MP-DocVQA, SlideVQA, and MMLongBench-Doc would also help establish a consistent reproduction baseline. For retrieval results, please specify the evaluated split/subset and the exact Top-1/Top-5 definition for questions with multiple evidence pages.

We have retained raw predictions, configurations, hashes, submission files, and per-category results, and can provide targeted examples or logs.

Our goal is to reproduce the published evaluation faithfully—not to tune submission formatting against hidden test scores. If some experimental settings differ from the public demo or checkpoint defaults, documenting those differences would be especially helpful.

Thank you!
